src/queue.js

```diff
@@ -25,17 +25,26 @@ export default class Queue {
         this.tail = node
       }
 
-      this.length += 1
-
+      this.size += 1
       return node
   }
 
   dequeue() {
-    
+    if (this.size === 0) return null
+    const node = this.head
+    this.head = node.next
+    node.next = null
+    this.head.previous = null
+    this.size -= 1
+    return node.data
   }
 
-  length() {
-    return this.length
-  }
+  front = () => this.size === 0 ? null : this.head.data
```

Comment:
I recommend consistency in code style within your codebase, and especially within a class.  These method definitions should be consistent with the method definitions above, e.g.:
```js
front() {
  return this.size === 0 ? null : this.head.getData()
}
```